src/v4/http/types.ts

```diff
@@ -137,7 +138,7 @@ export interface V4InteractionWithPlugin {
 
 export interface V4InteractionWithPluginRequest {
```

Comment:
Unrelated, but does this need to be a separate interface?

Reply:
It does, as the builder func is different and you must provide certain details prior to this being an allowed state. See the type-state pattern, which I think resolves so many problems with the current DSL (and with the more complicated options allowed in V4, such as plugins).